tests/forte/data/ontology/ontology_code_generator_test.py

```diff
@@ -77,105 +77,52 @@ def assert_generation_equal(self, file_a, file_b):
-            self.generator.generate(temp_filename, temp_dir, is_dry_run=False)
-            folder_path = temp_dir
-            for name in ["ft", "onto", "example_import_ontology.py"]:
-                self.assertTrue(name in os.listdir(folder_path))
-                folder_path = os.path.join(folder_path, name)
-
-    def test_include_and_exclude_init(self):
-        json_file_path = os.path.join(
-            self.spec_dir, "example_import_ontology.json"
-        )
-        with tempfile.TemporaryDirectory() as temp_dir:
-            temp_filename = _get_temp_filename(json_file_path, temp_dir)
-            # Test with include_init = True
+            # Test with include_init = True, namespace_depth = int
             folder_path = self.generator.generate(
-                temp_filename, temp_dir, is_dry_run=False, include_init=True
+                temp_filename,
+                temp_dir,
+                is_dry_run=False,
+                include_init=True,
+                namespace_depth=namespace_depth,
             )
             gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
 
-            # Assert the generated python files
-            exp_file_path = [
-                "ft/__init__",
-                "ft/onto/__init__",
-                "ft/onto/example_import_ontology",
+            exp_file_path_all = [
+                "ft/__init__.py",
+                "ft/onto/__init__.py",
+                "ft/onto/example_import_ontology.py",
             ]
+            exp_file_path = exp_file_path_all[namespace_depth:]
             exp_files = sorted(
                 [
-                    f"{os.path.join(folder_path, file)}.py"
+                    f"{os.path.join(folder_path, file)}"
                     for file in exp_file_path
                 ]
             )
 
             self.assertEqual(gen_files, exp_files)
 
-            # Now, corrupt one of the init files
-            corrupted_path = os.path.join(folder_path, "ft/__init__.py")
-            with open(corrupted_path, "w") as f:
-                f.write("# ***corrupted file***\n")
-
-            # Re-generate using include_init = False
+            # Re-generate using include_init = False, namespace_depth = int
             self.generator = OntologyCodeGenerator()
             folder_path = self.generator.generate(
-                temp_filename, folder_path, is_dry_run=False, include_init=False
+                temp_filename,
+                folder_path,
+                is_dry_run=False,
+                include_init=False,
+                namespace_depth=namespace_depth,
             )
             gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
 
-            # Assert the generated python files after removing the corrupted
-            # file which should not have been regenerated
-            exp_files = [file for file in exp_files if file != corrupted_path]
             self.assertEqual(gen_files, exp_files)
```

Comment:
Is it the case that the `exp_files` is the same no matter when `include_init = True` or `include_init = False`?  I think their behavior should be different. Do you need to remove the temporary files between different generation steps?

Reply:
Hi Hector,
1. the readme is updated, please kindly check __main__.py line 202 and 206-209
2. May I ask the priority of include_init and namespace_depth, as they are controlling the same behaviour at some generation steps. Currently, namespace_depth has a higher priority.

Reply:
1. I didn't see the readme got updated. I mean literally this file: https://github.com/asyml/forte/blob/master/docs/toc/ontology_generation.md, you can see here we list the options for the generation tool: https://github.com/asyml/forte/blob/master/docs/toc/ontology_generation.md#ontology-generation-steps, now since we update the tool, this part needs to be updated.
2. I see, so `include_init` might not be needed. So to clarify this, we have two options: a) remove the `include_init` b) explain in docstring on which one have priority, and what happen if we use one of them but not the other.   I would say removing `include_init` might be easier?